src/SegmentedMsg.chpl

```diff
@@ -180,72 +193,63 @@ module SegmentedMsg {
-            st.addEntry(rvname, new shared SymEntry(rv));
-          } otherwise {
-              var errorMsg = notImplementedError(pn, 
-                               "subcmd: %s, (%s, %s)".format(subcmd, objtype, valtype));
-              smLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);      
-              return new MsgTuple(errorMsg, MsgType.ERROR);                            
+
+          if regex {
+            var (lo, lv, ro, rv) = strings.peelRegex(val, times, includeDelimiter, keepPartial, left);
+            createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+          }
+          else {
+            select (includeDelimiter, keepPartial, left) {
+              when (false, false, false) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, false, false, false);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+              }
+              when (false, false, true) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, false, false, true);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+              }
+              when (false, true, false) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, false, true, false);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+              }
+              when (false, true, true) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, false, true, true);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+              }
+               when (true, false, false) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, true, false, false);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
               }
+              when (true, false, true) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, true, false, true);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+              }
+              when (true, true, false) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, true, true, false);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+              }
+              when (true, true, true) {
+                var (lo, lv, ro, rv) = strings.peel(val, times, true, true, true);
+                createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
+              }
+              otherwise {
+                var errorMsg = notImplementedError(pn,
+                                "subcmd: %s, (%s, %s)".format(subcmd, objtype, valtype));
+                smLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR);
+              }
+            }
```

Comment:
Hmm.. the more I look at this the more I am convinced it can be collapsed even further.  There are 3 boolean args (2 states, 3 times) which should mean a total of 2^3 permutations (aka 3 bits) = 8.  All 8 permutations are represented here so the `otherwise` clause is unreachable and not needed.  Since we are putting the boolean args straight into the peel function call _as is_ I think we can do away with the entire `select` and straight up call
```
var (lo, lv, ro, rv) = strings.peel(val, times, includeDelimiter, keepPartial, left);
createPeelSymEntries(loname, lo, lvname, lv, roname, ro, rvname, rv, st);
```

Reply:
I had the same thought when I first saw this. I did just enough looking into it to convince myself it's [some optimization thing with`param`s ]( https://chapel-lang.org/docs/users-guide/base/constParam.html#when-to-use-const-vs-param). I don't really understand it but I figure there had to be a reason to pass in the literal `true/false` instead of the `bool` variables. I didn't know enough to feel confident changing it. Maybe someone else can weigh in on why there's a switch statement on the `includeDelimiter`, `keepPartial`, and `left` `bool`s instead of just passing these variables to `peel`?

I do agree that `otherwise` is completely unnecessary and I'll go ahead and remove that